tendermint-testgen/src/vote.rs

```diff
@@ -0,0 +1,140 @@
+use gumdrop::Options;
+use serde::Deserialize;
+use simple_error::*;
+use tendermint::{
+    Time, vote, block, lite, signature::Signature };
+use signatory::{
+    ed25519,
+    signature::{ Signature as _, Signer }
+};
+
+use crate::{Generator, Validator, Header, helpers::*};
+
+#[derive(Debug, Options, Deserialize, Clone)]
+pub struct Vote {
+    #[options(help = "validator of this vote (required; can be passed via STDIN)",
+      parse(try_from_str = "parse_as::<Validator>"))]
+    pub validator: Option<Validator>,
+    #[options(help = "validator index (default: from commit header)")]
+    pub index: Option<u64>,
+    #[options(help = "header to sign (default: commit header)")]
+    pub header: Option<Header>,
+    #[options(help = "vote type; 'precommit' if set, otherwise 'prevote' (default)")]
+    pub precommit: Option<()>,
+    #[options(help = "block height (default: from header)")]
+    pub height: Option<u64>,
+    #[options(help = "time (default: from header)")]
+    pub time: Option<Time>,
+    #[options(help = "commit round (default: from commit)")]
+    pub round: Option<u64>,
+}
+
+impl Vote {
+    pub fn new(validator: &Validator, header: &Header) -> Self {
+        Vote {
+            validator: Some(validator.clone()),
+            index: None,
+            header: Some(header.clone()),
```

Comment:
Usually preferable to leave it up to the callee to clone the data passed to a function. This can avoid unnecessary clones in cases where the callee already has ownership of the data passed to the function.

```suggestion
    pub fn new(validator: Validator, header: Header) -> Self {
        Vote {
            validator: Some(validator),
            index: None,
            header: Some(header),
```